langserver/symbol.go

```diff
@@ -244,6 +254,21 @@ func (h *LangHandler) handleSymbol(ctx context.Context, conn JSONRPC2Conn, req *
 		par := parallel.NewRun(8)
 		pkgs := buildutil.ExpandPatterns(bctx, []string{pkgPat})
 		for pkg := range pkgs {
+			// If we're restricting results to a single file or dir, ensure the
+			// package dir matches to avoid doing unnecessary work.
+			if results.query.file != "" {
+				filePkgPath := path.Dir(results.query.file)
+				filePkgPath = PathTrimPrefix(filePkgPath, h.init.BuildContext.GOROOT)
```

Comment:
seems incorrect technically, like if it contained multiple of those prefixes strangely like `/usr/local/go/src/usr/local/go/src/src/gopath/src/github.com/foo/bar` it would still result in `github.com/foo/bar` although it should fail.